includes/utils/class-amp-dom-utils.php

```diff
@@ -64,8 +64,21 @@ public static function get_dom( $document ) {
 
 		$dom = new DOMDocument();
 
-		// @todo In the future consider an AMP_DOMDocument subclass that does this automatically. See <https://github.com/Automattic/amp-wp/pull/895/files#r163825513>.
-		$document = self::convert_amp_bind_attributes( $document );
+		/**
+		 * Convert Bind Attributes.
+		 *
+		 * Use this filter to skip converting amp-bind style attributes. On large documents that
+		 * are known not to contain amp-bind style attributes, it's more performant to skip the
+		 * conversion.  Use of this filter is only recommended if documents are very large and
+		 * at risk of exceeding regex engine limits.
+		 *
+		 * @param bool   $convert  True if bind attributes should be converted, false otherwise. Default true.
+		 * @param string $document Valid HTML document possibly containing bind attributes to be converted.
+		 */
+		if ( apply_filters( 'amp_convert_bind_attributes', true, $document ) ) {
+			// @todo In the future consider an AMP_DOMDocument subclass that does this automatically. See <https://github.com/Automattic/amp-wp/pull/895/files#r163825513>.
+			$document = self::convert_amp_bind_attributes( $document );
```

Comment:
Note that while the conversion is conditional here, the restoration is not conditional. Below you can see `restore_amp_bind_attributes` gets called even though no conversion may have been done. This isn't necessarily a problem, but it is a wasted search/replace operation.